src/Handler/StreamHandler.php

```diff
@@ -106,18 +107,30 @@ private function checkDecode(array $options, array $headers, $stream)
     {
         // Automatically decode responses when instructed.
         if (!empty($options['decode_content'])) {
-            foreach ($headers as $key => $value) {
-                if (strtolower($key) == 'content-encoding') {
-                    if ($value[0] == 'gzip' || $value[0] == 'deflate') {
-                        return new Psr7\InflateStream(
-                            Psr7\stream_for($stream)
-                        );
+            $normalizedKeys = array_combine(array_map('strtolower', array_keys($headers)), array_keys($headers));
+
+            if (isset($normalizedKeys['content-encoding'])) {
+                $encoding = $headers[$normalizedKeys['content-encoding']];
```

Comment:
After the `array_change_key_case` change, modify this to be something like `$encoding = strtolower($lcheaders['content-encoding'])`